### PR TITLE
Changes to finish recording on mobile session

### PR DESCRIPTION
**Pull request by jankrzempek**
What was done:

Finish recording in now available through action Menu.

https://trello.com/c/CEL1yPL5/186-mobile-active-buttons

### CODE REVIEW

File: AirCasting/AirCastingApp.swift [hunk 1]
```diff
@@ -15,6 +15,7 @@ struct AirCastingApp: App {
     private let authorization: UserAuthenticationSession
     private let syncScheduler: SynchronizationScheduler
     private let sessionSynchronizer: SessionSynchronizer
+    private let microphoneMenager: MicrophoneManager
```
Comment (pawelgil) [inline AirCasting/AirCastingApp.swift:18]: typo Menager -> Manager

File: AirCasting/MicrophoneSession/MicrophoneManager.swift [hunk 2]
```diff
@@ -25,9 +25,11 @@ final class MicrophoneManager: NSObject, ObservableObject {
     private var levelTimer: Timer?
     private(set) var session: Session?
     private lazy var locationProvider = LocationProvider()
+    let sessionSynchronizer: SessionSynchronizer
```
Comment (pawelgil) [inline AirCasting/MicrophoneSession/MicrophoneManager.swift:28]: Good thing you used a `protocol` here, but overall I think this class is a huge dump of logic that should really be handled elsewhere. It's usually the case with objects named `Manager` ;) Let's leave it like this for now, but in the future try to find solutions that doesn't touch those classes :)

File: AirCasting/SessionViews/SessionHeaderView.swift [hunk 3]
```diff
@@ -73,8 +78,37 @@ private extension SessionHeaderView {
         }
         .foregroundColor(.darkBlue)
     }
-
-    var actionsMenu: some View {
+    
+    var actionsMenuMobile: some View {
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionHeaderView.swift:82]: this got huge, let's split stuff into helpers `var/func`

File: AirCasting/SessionViews/SessionHeaderView.swift [hunk 4]
```diff
@@ -73,8 +78,37 @@ private extension SessionHeaderView {
...
+        } label: {
+            ZStack(alignment: .trailing) {
+                EditButtonView()
+                Rectangle()
+                    .frame(width: 30, height: 20, alignment: .trailing)
+                    .opacity(0.0001)
+            }
+        }.alert(isPresented: $showingFinishAlert) {
+            Alert(title: Text(Strings.SessionHeaderView.finishAlertTitle) +
+                Text(session.name ?? "this session")
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionHeaderView.swift:98]: Strings

File: AirCasting/SessionViews/StreamsView.swift [hunk 5]
```diff
@@ -34,7 +30,7 @@ struct StreamsView: View {
             Text(Strings.SessionCart.measurementsTitle)
                 .font(Font.moderate(size: 12))
                 .padding(.bottom, 3)
-                .padding(.horizontal)
+//                .padding(.horizontal)
```
Comment (pawelgil) [inline AirCasting/SessionViews/StreamsView.swift:33]: Don't leave commented out code

File: AirCasting/CoreData/MeasurementStreamStorage.swift [hunk 6]
```diff
@@ -100,6 +101,19 @@ final class CoreDataMeasurementStreamStorage: MeasurementStreamStorage {
         sessionEntity.status = sessionStatus
         try context.save()
     }
+    
+    func updateSessionEndTime(for sessionUUID: SessionUUID) throws {
+        let context = persistenceController.editContext()
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        sessionEntity.endTime = Date()
```
Comment (pawelgil) [inline AirCasting/CoreData/MeasurementStreamStorage.swift:108]: into the perform block it shall go

File: AirCasting/CoreData/MeasurementStreamStorage.swift [hunk 7]
```diff
@@ -100,6 +101,19 @@ final class CoreDataMeasurementStreamStorage: MeasurementStreamStorage {
         sessionEntity.status = sessionStatus
         try context.save()
     }
+    
+    func updateSessionEndTime(for sessionUUID: SessionUUID) throws {
+        let context = persistenceController.editContext()
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
```
Comment (pawelgil) [inline AirCasting/CoreData/MeasurementStreamStorage.swift:107]: into the perform block it shall go

**Review by pawelgil — CHANGES_REQUESTED** [review]

File: AirCasting/SessionViews/StreamsView.swift (hunk 5 above)
Comment (jankrzempek) [reply AirCasting/SessionViews/StreamsView.swift:33]: Of course!

**Review by pawelgil — APPROVED** [review]

**Pull request merged**